pandas/_libs/tslibs/offsets.pyx

```diff
@@ -2014,7 +2014,7 @@ cdef class BusinessHour(BusinessMixin):
             nb_offset = 1
         else:
             nb_offset = -1
-        if self._prefix.startswith("C"):
+        if self._prefix.startswith(("C", "c")):
             # CustomBusinessHour
             return CustomBusinessDay(
```

Comment:
should this one check for both? if it's business hour, don't we know it's going to start with `'c'`?

Reply:
Thank you for reviewing this PR.

I think we need to check both: upper and lowercase 'c', because we added `"cbh"` to the dictionary _`dont_uppercase = {"h", "bh", "cbh", "MS", "ms", "s", "me"}.` 
On the other hand, if freq is not in  `_dont_uppercase`  we do uppercase in `_get_offset()`.

When I remove the check for "c", tests for CustomBusinessHour fail.

Reply:
thanks for explaining - the comment 
```python
            # CustomBusinessHour
            return CustomBusinessDay
```
might be a bit misleading, but I'll pull your code tomorrow, step through it, and check

the rest looks good though, so we should be very close

Reply:
Thanks, I agree, checking `if self._prefix.startswith("c"):` should be enough. I changed the condition, locally all tests passed.